### PR TITLE
update mimir-prometheus at grafana/mimir-prometheus@53cf3fb

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Update mimir-prometheus at https://github.com/grafana/mimir-prometheus/commit/53cf3fb8e7e3c89e4f78b0ab0b1468899d8da177

Includes fixes we depend on:
https://github.com/prometheus/prometheus/pull/14771
https://github.com/prometheus/prometheus/pull/14831

Pin Google GRPC to 1.65, ref https://github.com/grafana/dskit/pull/581
Keep pinning client_golang to 1.19.1 due to data races in histogram exemplars. https://github.com/prometheus/client_golang/pull/1623 , https://github.com/prometheus/client_golang/pull/1608

The remote read client interface has changed in https://github.com/prometheus/prometheus/pull/11379
I've updated the users and also added native histogram support to backfill.

The client golang update contains https://github.com/prometheus/client_golang/pull/1424 which invalidates some
tests and also exposes some test errors. From now on `GatherAndCompare` checks if all supplied metrics are actually
present and errors otherwise, this it cannot be used to assert on missing metrics as before.

### CODE REVIEW

File: pkg/distributor/otel.go [hunk 1]
```diff
@@ -429,7 +429,7 @@ func otelMetricsToTimeseries(ctx context.Context, tenantID string, addSuffixes b
 // Old, less efficient, version of otelMetricsToTimeseries.
 func otelMetricsToTimeseriesOld(ctx context.Context, tenantID string, addSuffixes bool, discardedDueToOtelParseError *prometheus.CounterVec, logger log.Logger, md pmetric.Metrics) ([]mimirpb.PreallocTimeseries, error) {
 	converter := prometheusremotewrite.NewPrometheusConverter()
-	errs := converter.FromMetrics(ctx, md, prometheusremotewrite.Settings{
+	_, errs := converter.FromMetrics(ctx, md, prometheusremotewrite.Settings{
```
Comment (krajorama) [inline pkg/distributor/otel.go:432]: note to reviewers: I'm not sure what we'd do with annotations here @aknuds1

File: pkg/ruler/remotequerier.go [hunk 2]
```diff
@@ -195,6 +197,17 @@ func (q *RemoteQuerier) Read(ctx context.Context, query *prompb.Query) (*prompb.
...
 	level.Debug(log).Log("msg", "remote read successfully performed", "qs", query)
 
+	var contentType string
+	for _, h := range resp.GetHeaders() {
+		if strings.ToLower(h.GetKey()) == "content-type" {
+			contentType = h.GetValues()[0]
+			break
+		}
+	}
+	if contentType != "application/x-protobuf" {
```
Comment (krajorama) [inline pkg/ruler/remotequerier.go:207]: note to reviewers: I'm not sure this is needed here

**Comment by krajorama** [comment]
@aknuds1 I'm ignoring annotations coming from OTLP metrics conversion for now , you might want to change that later.

File: pkg/distributor/otel.go (hunk 1 above)
Comment (jesusvazquez) [reply pkg/distributor/otel.go:432]: You can do this:
```golang
	annots, errs := converter.FromMetrics(ctx, md, prometheusremotewrite.Settings{
		AddMetricSuffixes:                   addSuffixes,
		EnableCreatedTimestampZeroIngestion: enableCTZeroIngestion,
	})
	promTS := converter.TimeSeries()
	if errs != nil {
                ...
		level.Warn(logger).Log("msg", "OTLP parse error", "err", parseErrs)
	}
	ws, _ := annots.AsStrings("", 0, 0)
	if len(ws) > 0 {
		level.Warn(logger).Log("msg", "Warnings translating OTLP metrics to Prometheus write request", "warnings", ws)
	}
```

Comment (krajorama) [reply pkg/distributor/otel.go:432]: followup PR? there's already a lot going on in this PR

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/mimir
 
-go 1.22
+go 1.22.0
```
Comment (pstibrany) [inline go.mod:3]: what effect does this have?

**Review by pstibrany — APPROVED** [review]

File: go.mod (hunk 3 above)
Comment (pstibrany) [reply go.mod:3]: From reading https://go.dev/doc/toolchain, I don't think there's any difference between the two.

**Review by jesusvazquez — APPROVED** [review]
LGTM 👍 

Also checked that this brings https://github.com/grafana/mimir-prometheus/pull/689

File: go.mod (hunk 3 above)
Comment (krajorama) [reply go.mod:3]: I'm not sure, `go mod tidy` sets it

**Pull request merged**